### DISCUSSION TITLE
How to exclude file from linting?

### DISCUSSION THREAD

**Discussion by AloisSeckar** [nuxt/eslint-config#413] [Q&A] [answered]
I have a file `/utils/_api.ts` auto-generated by OpenAPI containing TS types from backend. It doesn't fit ESLint rules, but since it is auto-generated, I just want to ignore it. However, I cannot figure out, how to make it work.

Based on my older solution with `ignorePatterns` and after studing the flat config docs I belive, in `eslint.config.mjs` I should have:

```js
export default withNuxt({
  ignores: ['**/_api.ts']
  // other config
})
```

but the file is still being checked and linting issues reported. What do I do wrong? Overriding `rules` section like this works good.

**Comment by dsvgl** [comment, marked as answer]
I also had problems with ignoring. I have a separate package with separate eslint configs.
I have a dedicated config for when it should be used inside `withNuxt` (aka the package/app is a nuxt app).

```js
export const companyIgnores = {
  // NOTE: no "name" else, it wouldn't work; maybe a bug in ESLint
  ignores: ['**/node_modules', '**/public', '**/vendor', '**/dist', '**/.nuxt'],
};

export const configNuxt = [eslintPluginPrettierRecommended, companyEslintRules, companyIgnores];
```

This works for me. Could you try making entry an array?

```js
export default withNuxt([
  {
    ignores: ['**/_api.ts'],
  },
  // other config
]);
```

You could also try to prepend the ignores befor the nuxt rules:
```js
export default withNuxt(
  // ...Custom flat configs append after nuxt's configs
).prepend(
  // ...Prepend some flat configs in front
)
```

see https://eslint.nuxt.com/packages/module#config-customizations

**Comment by lmc3s3f0r** [comment]
Having the same proble here. My eslint.config.mjs it this:

import withNuxt from './.nuxt/eslint.config.mjs'

```
export default withNuxt({
  ignores: ['assets/primevueThemes/**/*.js'],
  rules: {
    '@typescript-eslint/no-explicit-any': 'off',
    'vue/multi-word-component-names': 'off',
    'vue/prop-name-casing': 'off',
    'vue/html-self-closing': 'off',
    'no-console': ['error', { allow: ['warn', 'error', 'debug', 'table', 'info'] }],
    camelcase: 'off'
  }
})
```

The file is detected, as I tried removing the "camelcase" line and it reacted, but "ignores" is not working.

Thanks to @dsvgl I learnt that you can define config as an array of objects, so I changed it to this:

```
import withNuxt from './.nuxt/eslint.config.mjs'

export default withNuxt([
  {ignores: ['assets/primevueThemes/**/*.js']},
  {rules: {
    '@typescript-eslint/no-explicit-any': 'off',
    'vue/multi-word-component-names': 'off',
    'vue/prop-name-casing': 'off',
    'vue/html-self-closing': 'off',
    'no-console': ['error', { allow: ['warn', 'error', 'debug', 'table', 'info'] }],
    camelcase: 'off'
  }}
])
```

And now it works. Thanks a lot @dsvgl !

**Comment by AloisSeckar** [comment]
Thank you @dsvgl, the trick with an array of objects does to job for me. However, I cannot help myself but to consider it ugly and counter-intuitive. Why is not possible to use just one compact config object?

**Reply by dsvgl** [reply]
Also curious why this doesn’t work
